ibm/resource_ibm_network_gateway.go

```diff
@@ -0,0 +1,996 @@
+			Domain:   sl.String(d.Get("domain").(string)),
+		},
+		},
+		Location:  sl.String(strconv.Itoa(*dc.Id)),
+		PackageId: pkg.Id,
+		Prices: []datatypes.Product_Item_Price{
+			server,
+			os,
+			ram,
+			portSpeed,
+			priIpAddress,
+			pri_ipv6_addresses,
+			remoteManagement,
+			vpnManagement,
+			monitoring,
+			notification,
+			response,
+			vulnerabilityScanner,
+		},
+	}
+
+	// Add optional price ids.
+	// Add public bandwidth
+	publicBandwidth, ok := d.GetOk("public_bandwidth")
+	bandwidth, err := getItemPriceIdGateway(items, "bandwidth", publicBandwidth.(string))
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	order.Prices = append(order.Prices, bandwidth)
+
+	// Add prices of disks.
+	var arrayDrives []interface{}
+	var arrayDrivestemp []string
+	arrayDrivestemp = append(arrayDrivestemp, "HARD_DRIVE_1_00_TB_SATA_III")
+	arrayDrivestemp = append(arrayDrivestemp, "HARD_DRIVE_2_00TB_SATA_II")
+
+	for _, val := range arrayDrivestemp {
+		arrayDrives = append(arrayDrives, val)
+	}
+
+	diskLen := len(arrayDrives)
+	if diskLen > 0 {
+		for i, disk := range arrayDrives {
+			diskPrice, err := getItemPriceIdGateway(items, "disk"+strconv.Itoa(i), disk.(string))
+			if err != nil {
+				return datatypes.Container_Product_Order{}, err
+			}
+			order.Prices = append(order.Prices, diskPrice)
+		}
+	}
+
+	// Add storage_groups for RAID configuration
+	//hard coded support for disk controller, just RAID 1 is supported in this release of the resource
+	diskController, err := getItemPriceIdGateway(items, "disk_controller", "DISK_CONTROLLER_RAID_1")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	order.Prices = append(order.Prices, diskController)
+
+	return order, nil
+}
+func getPackageByModelGateway(sess *session.Session, model string) (datatypes.Product_Package, error) {
+	objectMask := "id,keyName,name,description,isActive,type[keyName],categories[id,name,categoryCode]"
+	service := services.GetProductPackageService(sess)
+	availableModels := ""
+	filterStr := "{\"items\": {\"categories\": {\"categoryCode\": {\"operation\":\"server\"}}},\"type\": {\"keyName\": {\"operation\":\"BARE_METAL_GATEWAY\"}}}"
+
+	// Get package id
+	packages, err := service.Mask(objectMask).
+		Filter(filterStr).GetAllObjects()
+	if err != nil {
+		return datatypes.Product_Package{}, err
+	}
+	for _, pkg := range packages {
+		availableModels = availableModels + *pkg.KeyName
+		if pkg.Description != nil {
+			availableModels = availableModels + " ( " + *pkg.Description + " ), "
+		} else {
+			availableModels = availableModels + ", "
+		}
+		if *pkg.KeyName == model {
+			return pkg, nil
+		}
+	}
+	return datatypes.Product_Package{}, fmt.Errorf("No Gateway package key name for %s. Available package key name(s) is(are) %s", model, availableModels)
+}
+
+func getItemPriceIdGateway(items []datatypes.Product_Item, categoryCode string, keyName string) (datatypes.Product_Item_Price, error) {
```

Comment:
Can't we re-use the `getItemPriceId` from `ibm_bare_metal` implementation?

Reply:
All instances of getItemPriceIdGateway have been changed to use getItemPriceId